nomos-da/network/core/src/test_utils.rs

```diff
@@ -27,3 +27,35 @@ impl MembershipHandler for AllNeighbours {
         self.neighbours.clone()
     }
 }
+
+use crossbeam_channel::{bounded, Receiver, Sender};
+
+/// A special-purpose multi-producer, multi-consumer(MPMC) channel to relay messages indicating whether the associated stream should be closed or not.
```

Comment:
Hm..I think I could've first clarified a bit about the usage...<br>
### Oneshot
Suppose `peer1` has finished sampling but the data packet for the last sample has not reached to `peer2`.
According to the `SamplingBehaviour`, on the very next iteration of `peer2` the `IncomingSample` event will be triggered.
But in this same loop, the `peer2` will try to listen to peer1's indicator channel. If you make the channel *oneshot*, then you would not be able to listen to peer1 in the next iteration (where you will actually get the sample in hand i.e. the `SamplingSuccess`) as you can only send message once.
### Tokio::MPSC
I preffered **crossbeam::mpmc** over **tokio::mpsc** in view of the potential need of sending messages across threads.
In future, if I need to send message across two async runtime threads( let's say t1 and t2) then tokio::mpsc is unable to do that.. because in the receiver in mpsc can listen **only in the thread which it was created**. Besides, crossbeam's mpmc is actually a wrapper over mpsc, so there is really no overhead.